### ISSUE TITLE
with Amplify 2.+, recall confirmSignin after typing wrong code the first time throws InvalidStateException

### ISSUE THREAD

**Issue by sokarcreative** [aws-amplify/amplify-android#2212]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
implementation("com.amplifyframework:core-kotlin:2.1.0")
implementation("com.amplifyframework:aws-auth-cognito:2.1.0")
```

</details>

### Environment information

<details>

```
Kotlin:       1.5.31
Groovy:       3.0.9
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          16.0.1 (AdoptOpenJDK 16.0.1+9)
OS:           Mac OS X 10.16 x86_64
```

</details>

### Describe the bug

At signed out state, after being signed in with email and password successfully, If I'm confirming SignIn with a wrong code the first time, I'm getting an InvalidStateException the next time I'm confirming signIn wether or not it's a valid code.

It worked fine in pre 2.0.0.

### Reproduction steps (if applicable)

In order, when I do :
1. Fresh install or signout
2. signIn with email password valid with successfull response
At this point an sms with code has been sent
3. perform confirmSignIn with wrong code -> CodeMismatchException{message=Confirmation code entered is not correct...
4. perform confirmSignin with right code ->  InvalidStateException{message=Auth state is an invalid state, cannot process the request., cause=null, recoverySuggestion=Operation performed is not a valid operation for the current auth state.}

### Log output

<details>

```
10:41:39.117 [UC] amplifySignIn   V  [UC] START
10:41:39.118 [Amplify]            I  calling.. signIn
10:41:41.829 [Amplify]            V  signIn succeed
10:41:41.830 [UC] amplifySignIn   V  [UC] END succeed 
10:42:01.034 [UC] amplifyConfirmSignIn  V  [UC] START
10:42:01.034 [Amplify]            I  calling.. confirmSignIn
10:42:01.223 [Amplify]            W  confirmSignIn error -> CodeMismatchException{message=Confirmation code entered is not correct., cause=CodeMismatchException(message=Invalid code or auth state for the user.), recoverySuggestion=Enter correct confirmation code.}
                                         CodeMismatchException{message=**Confirmation code entered is not correct.**, cause=CodeMismatchException(message=Invalid code or auth state for the user.), recoverySuggestion=Enter correct confirmation code.}
10:42:01.224 [UC] amplifyConfirmSignIn  W  [UC] END error -> CodeMismatchException{message=Confirmation code entered is not correct., cause=CodeMismatchException(message=Invalid code or auth state for the user.), recoverySuggestion=Enter correct confirmation code.}
                                         CodeMismatchException{message=Confirmation code entered is not correct., cause=CodeMismatchException(message=Invalid code or auth state for the user.), recoverySuggestion=Enter correct confirmation code.}
10:42:25.790 [UC] amplifyConfirmSignIn V  [UC] START
10:42:25.790 [Amplify]            I  calling.. confirmSignIn
10:42:25.796 [Amplify]            W  confirmSignIn error -> InvalidStateException{message=Auth state is an invalid state, cannot process the request., cause=null, recoverySuggestion=Operation performed is not a valid operation for the current auth state.}
                                         InvalidStateException{message=Auth state is an invalid state, cannot process the request., cause=null, recoverySuggestion=Operation performed is not a valid operation for the current auth state.}
10:42:25.796 [UC] amplifyConfirmSignIn W  [UC] END error -> InvalidStateException{message=Auth state is an invalid state, cannot process the request., cause=null, recoverySuggestion=Operation performed is not a valid operation for the current auth state.}
                                         InvalidStateException{message=Auth state is an invalid state, cannot process the request., cause=null, recoverySuggestion=Operation performed is not a valid operation for the current auth state.}
```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by sokarcreative** [comment]
Come on, this bug is huge and still no PR on it...
Why create a major version 2.x.y with unstable okhttp3 alpha dependency and some big issues with signIn.
No sense.

**Comment by gpanshu** [comment]
This has now been fixed and available in Amplify Android V 2.2.2.
